test/extended/router/gatewayapicontroller.go

```diff
@@ -0,0 +1,504 @@
+				o.Expect(errIstio).NotTo(o.HaveOccurred())
+				if istioStatus != "Healthy" {
+					e2e.Logf("Istio CR %q is not healthy, retrying...", resource.Name)
+					return false, nil
+				}
+				e2e.Logf("Istio CR %q is healthy", resource.Name)
+				return true, nil
+			})
+			o.Expect(waitCR).NotTo(o.HaveOccurred(), "Istio CR %q did not reach healthy state in time", resource.Name)
+
+		})
+		g.It("and ensure default gatewayclass is accepted", func() {
+
+			g.By("Check if default GatewayClass is accepted after OLM resources are successful")
+			errCheck := checkGatewayClass(oc, gatewayClassName)
+			o.Expect(errCheck).NotTo(o.HaveOccurred())
+			e2e.Logf("GatewayClass %s successfully installed and accepted!", gatewayClassName)
+
+		})
+		g.It("and ensure custom gatewayclass can be accepted", func() {
+			gwapiClient := gatewayapiclientset.NewForConfigOrDie(oc.AdminConfig())
+
+			g.By("Create Custom GatewayClass")
+			gatewayClass := buildGatewayClass("custom-gatewayclass", gatewayClassControllerName)
+			gwc, err := gwapiClient.GatewayV1().GatewayClasses().Create(context.TODO(), gatewayClass, metav1.CreateOptions{})
+			if err != nil {
+				e2e.Logf("Gateway Class \"custom-gatewayclass\" already exists, or has failed to be created, checking its status")
+			}
+			errCheck := checkGatewayClass(oc, "custom-gatewayclass")
+			o.Expect(errCheck).NotTo(o.HaveOccurred())
+			e2e.Logf("GatewayClass %s successfully installed and accepted!", gwc.Name)
+
+			g.By("Deleting Custom GatewayClass and confirming that it is no longer there")
+			err = gwapiClient.GatewayV1().GatewayClasses().Delete(context.Background(), "custom-gatewayclass", metav1.DeleteOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			_, err = gwapiClient.GatewayV1().GatewayClasses().Get(context.Background(), "custom-gatewayclass", metav1.GetOptions{})
+			o.Expect(err).To(o.HaveOccurred(), "The custom gatewayClass \"custom-gatewayclass\" has been sucessfully deleted")
+
+			g.By("check if default gatewayClass is accepted and ISTIO CR and pod are still available")
+			defaultCheck := checkGatewayClass(oc, gatewayClassName)
+			o.Expect(defaultCheck).NotTo(o.HaveOccurred())
+
+			g.By("Confirm that ISTIO CR is created and in healthy state")
+			resource := types.NamespacedName{Namespace: "openshift-ingress", Name: "openshift-gateway"}
+
+			istioStatus, errIstio := oc.AsAdmin().Run("get").Args("-n", resource.Namespace, "istio", resource.Name, "-o=jsonpath={.status.state}").Output()
+			o.Expect(errIstio).NotTo(o.HaveOccurred())
+			o.Expect(istioStatus).To(o.Equal(`Healthy`))
+		})
+
+		g.It("and ensure default gateway objects is created", func() {
+			g.By("Getting the default domain")
+			defaultIngressDomain, err := getDefaultIngressClusterDomainName(oc, time.Minute)
+			o.Expect(err).NotTo(o.HaveOccurred(), "failed to find default domain name")
+			defaultDomain := strings.Trim(defaultIngressDomain, "apps.")
```

Comment:
The code `defaultDomain := strings.Trim(defaultIngressDomain, "apps.")`  will remove all character like "s" in defaultIngressDomain, suggest to update it to: `defaultDomain := strings.Split(defaultIngressDomain, "apps.")[1]`

Reply:
thanks